kinto_remote_settings/signer/__init__.py

```diff
@@ -0,0 +1,277 @@
+        for_resources=("collection",),
+    )
+
+    config.add_subscriber(
+        functools.partial(
+            listeners.create_editors_reviewers_groups,
+            resources=resources,
+            editors_group=global_settings["editors_group"],
+            reviewers_group=global_settings["reviewers_group"],
+        ),
+        ResourceChanged,
+        for_actions=(ACTIONS.CREATE,),
+        for_resources=("collection",),
+    )
+
+    config.add_subscriber(
+        functools.partial(listeners.cleanup_preview_destination, resources=resources),
+        ResourceChanged,
+        for_actions=(ACTIONS.DELETE,),
+        for_resources=("collection",),
+    )
+
+    config.add_subscriber(
+        functools.partial(listeners.prevent_collection_delete, resources=resources),
+        ResourceChanged,
+        for_actions=(ACTIONS.DELETE,),
+        for_resources=("collection",),
+    )
+
+    if not asbool(settings.get("signer.allow_floats", False)):
+        config.add_subscriber(
+            functools.partial(listeners.prevent_float_value, resources=resources),
+            ResourceChanged,
+            for_actions=(ACTIONS.CREATE, ACTIONS.UPDATE),
+            for_resources=("record",),
+        )
+
+    sign_data_listener = functools.partial(
+        listeners.sign_collection_data, resources=resources, **global_settings
+    )
+
+    # If StatsD is enabled, monitor execution time of listener.
+    if config.registry.statsd:
+        # Due to https://github.com/jsocol/pystatsd/issues/85
+        for attr in ("__module__", "__name__"):
+            origin = getattr(listeners.sign_collection_data, attr)
+            setattr(sign_data_listener, attr, origin)
+
+        statsd_client = config.registry.statsd
+        key = "plugins.signer"
+        sign_data_listener = statsd_client.timer(key)(sign_data_listener)
+
+    config.add_subscriber(
+        sign_data_listener,
+        ResourceChanged,
+        for_actions=(ACTIONS.CREATE, ACTIONS.UPDATE),
+        for_resources=("collection",),
+    )
+
+    def on_new_request(event):
+        """Send the kinto-signer events in the before commit hook.
+        This allows database operations done in subscribers to be automatically
+        committed or rolledback.
+        """
+        # Since there is one transaction per batch, ignore subrequests.
+        if hasattr(event.request, "parent"):
+            return
+        current = transaction.get()
+        current.addBeforeCommitHook(listeners.send_signer_events, args=(event,))
+
+    config.add_subscriber(on_new_request, NewRequest)
+
+    try:
+        from kinto_emailer import send_notification
+        from pyramid_mailer import IMailer
+
+        if config.registry.queryUtility(IMailer):
+            config.add_subscriber(send_notification, ReviewRequested)
+            config.add_subscriber(send_notification, ReviewApproved)
+            config.add_subscriber(send_notification, ReviewRejected)
```

Comment:
This is one area where this signer package drifts from [kinto-signer](https://github.com/Kinto/kinto-signer). We want to set up email notifications for signer events in production, but for unit tests, we don't want to register these notifications.

Reply:
Update: this setup now happens in the top-level `kinto_remote_settings` `includeme` in https://github.com/mozilla-services/kinto-dist/pull/1934/commits/40d8a2f76beef36231bfd18c05d6838f83fa790c